### ISSUE TITLE
elasticsuite cannot handle upper case store codes

### ISSUE THREAD

**Issue by ooxi** [Smile-SA/elasticsuite#527]
While Magento does allow upper case characters in store codes, elastic search forbids them in index names. Since elasticsuit uses the store code without any transformation in the index name, this leads to being unable to reindex Magento.

```
Catalog Search indexer process unknown error:
invalid_index_name_exception: Invalid index name [magento2_de_DE_catalog_product_20170907_174559], must be lowercase
ElasticSuite Category Indexing indexer process unknown error:
invalid_index_name_exception: Invalid index name [magento2_de_DE_catalog_category_20170907_174559], must be lowercase
ElasticSuite Thesaurus Indexing indexer process unknown error:
invalid_index_name_exception: Invalid index name [magento2_de_DE_thesaurus_20170907_174559], must be lowercase
```

larasearch had the same issue, but it was fixed there a while ago in [issue 43](https://github.com/iverberk/larasearch/issues/43).

This issue can be fixed fairly easy by patching [](https://github.com/Smile-SA/elasticsuite/blob/master/src/module-elasticsuite-core/Helper/IndexSettings.php#L108).

### Preconditions

Ubuntu Linux 17.04

Magento Version : 2.1.8

ElasticSuite Version : 2.3.6 (^2.3)

Environment : Developer

**Edit:** See pull request #528

**Comment by afoucret** [comment]
PR have been merged into branches master, 2.3.x and 2.4.x

**Comment by ooxi** [comment]
Thank you!
